### PR TITLE
Convert verbs provider to new logging.

### PR DESCRIPTION
**Pull request by pmmccorm**
Convert sockets provider to new common logging.
Convert psm provider to new common logging.

I tried to match the existing semantics of the individual logging
implementations. To enable all current logging export: FI_LOG_LEVEL=2

Signed-off-by: pmmccorm <patrick.m.mccormick@intel.com>

### CODE REVIEW

**Comment by pmmccorm** [comment]
@j-xiong @jithinjosepkl Do these look OK?

**Comment by j-xiong** [comment]
Looks fine to me for the psm part.

File: prov/sockets/src/sock_fabric.c [hunk 1]
```diff
@@ -46,6 +46,8 @@ const char sock_fab_name[] = "IP";
 const char sock_dom_name[] = "sockets";
 const char sock_prov_name[] = "sockets";
 
+useconds_t sock_progress_thread_wait = 0;
+
```
Comment (shefty) [inline prov/sockets/src/sock_fabric.c]: unrelated change

**Comment by jithinjosepkl** [comment]
Thanks @pmmccorm. Sockets provider changes look good.
